src/MongoDB.Driver.Core/Core/Authentication/ScramShaAuthenticator.cs

```diff
@@ -189,22 +193,72 @@ private class ClientFirst : ISaslStep
                 _hi = hi;
                 _hmac = hmac;
                 _rPrefix = rPrefix;
+                _baseThreadId = Thread.CurrentThread.ManagedThreadId;
+                _baseCulture = CultureInfo.CurrentCulture.ToString();
             }
 
             public byte[] BytesToSendToServer => _bytesToSendToServer;
 
             public bool IsComplete => false;
 
+            string BuildErrorMessage(
+                string baseMessage,
```

Comment:
Not sure that we need to have all of these fields, but I think, there is no harm in having them too